docs/src/main/sphinx/connector/iceberg.rst

```diff
@@ -348,6 +348,310 @@ and a file system location of ``/var/my_tables/test_table``::
         partitioning = ARRAY['c1', 'c2'],
         location = '/var/my_tables/test_table')
 
+.. _iceberg-special-tables:
+
+Special tables
+--------------
+
+The Iceberg connector maintains several hidden tables that provide metadata
+for a specific table. You can query each metadata table by appending the
+metadata table name to the table name::
+
+   SELECT * FROM "test_table$data"
+
+``$data`` table
+^^^^^^^^^^^^^^^
+
+The ``$data`` table is an alias for the Iceberg table.
+
+The statement::
+
+    SELECT * FROM "test_table$data"
+
+is equivalent to::
+
+    SELECT * FROM test_table
+
+``$properties`` table
+^^^^^^^^^^^^^^^^^^^^^
+
+The ``$properties`` table provides general metadata information about the table.
```

Comment:
Nit: Possibly the usage of `provides general metadata information about the table` is redundant without further clarifying that the metadata is user supplied (or generated) tags for the tables in addition to configured table properties (some of which are specific to iceberg).

Maybe `The $properties table provides access to general information about iceberg table configuration and any additional metadata key/value pairs that users or engines mighthave tagged the table with. `?

Alternatively, it might be ok to simply reference that this is similar to hive's `TBLPROPERTIES`?

For reference I found the following in [Hive's documentation](https://cwiki.apache.org/confluence/display/Hive/LanguageManual+DDL#LanguageManualDDL-CreateTable):
```
The TBLPROPERTIES clause allows you to tag the table definition with your own
metadata key/value pairs. Some predefined table properties also exist,
such as last_modified_user and last_modified_time which are automatically
added and managed by Hive.
```

Iceberg doesn't supply `last_modified_time` as a tblproperty itself, but we also don't specifically try to stop HMS from adding these.

Reply:
I just noticed there is another PR specifically related to `$properties`, so I'll likely move this comment there if appropriate =)

Reply:
> I just noticed there is another PR specifically related to $properties

[#10480](https://github.com/trinodb/trino/pull/10480) is already merged.
I'm updating correspondingly the docs here.